packages/fiori/src/SideNavigationItem.ts

```diff
@@ -272,6 +272,14 @@ class SideNavigationItem extends SideNavigationSelectableItemBase {
 	_toggle() {
 		if (this.items.length) {
 			this.expanded = !this.expanded;
+
+			if (this.items.some(item => item.selected)) {
```

Comment:
This is not working when an item is expanded/collapsed via keyboard. Just modify `get _selected` property to:

```
get _selected() {
		if (this.sideNavCollapsed || !this.expanded) {
			return this.selected || this.items.some(item => item.selected);
		}

		return this.selected;
	}
```

as we have similar logic when the whole side nav is collapsed.